tests/unit/Utilities/001LocalIndexing_UnitTest.cpp

```diff
@@ -0,0 +1,144 @@
+    }
+
+    // Random mix of DoF types.
+    mesh.useDefaultDGBasisFunctions(2);
+    mesh.useDefaultConformingBasisFunctions(3, 1);
+    mesh.useDefaultDGBasisFunctions(0, 2);
+
+    // Validate with empty unknowns
+    Utilities::ElementLocalIndexing indexing(numberOfUnknowns);
+    INFO("Initial index should be empty");
+    REQUIRE(indexing.getIncludedUnknowns().empty() == true);
+    for (const Base::Element* element : mesh.getElementsList()) {
+        indexing.reinit(element);
+        INFO("Element should match the reinit");
+        REQUIRE(indexing.getElement() == element);
+        indexing.validate();
+    }
+
+    // Switch unknowns and revalidate
+    std::vector<std::size_t> unknowns(1, 1);
+    indexing.reinit(unknowns);
+    indexing.validate();
+    for (const Base::Element* element : mesh.getElementsList()) {
+        indexing.reinit(element);
+        indexing.validate();
+    }
+
+    // Verify with local data
+    unknowns.resize(3);
+    unknowns[0] = 0;
+    unknowns[1] = 1;
+    unknowns[2] = 2;
+    indexing.reinit(unknowns);
+    INFO("Included unknowns should match reinit");
+    REQUIRE(indexing.getIncludedUnknowns() == unknowns);
+    for (const Base::Element* element : mesh.getElementsList()) {
+        indexing.reinit(element);
+        INFO("Check total basis function count");
+        REQUIRE(indexing.getNumberOfDoFs() ==
+                element->getTotalNumberOfBasisFunctions());
+        INFO("Offset for first unknown should be zero")
+        REQUIRE(indexing.getDoFOffset(0) == 0);
+        std::size_t dofZeroSize = element->getNumberOfBasisFunctions(0);
+
+        INFO("Correct number of DoFs for unknown 0");
+        REQUIRE(indexing.getNumberOfDoFs(0) == dofZeroSize);
+
+        INFO("Only the first unknown DoFs are before those of the second one");
+        REQUIRE(indexing.getDoFOffset(1) == dofZeroSize);
+        // More can be added if desired
+    }
+
+    // Check with partial unknowns, dropping the first unknown
+    unknowns.resize(2);
+    unknowns[0] = 1;
+    unknowns[1] = 2;
+    indexing.reinit(unknowns);
+    for (const Base::Element* element : mesh.getElementsList()) {
+        indexing.reinit(element);
+        INFO("First included unknown should have offset 0");
+        REQUIRE(indexing.getDoFOffset(1) == 0);
+        std::size_t dof1Size = element->getNumberOfBasisFunctions(1);
+        INFO("Correct number of DoFs for first unknown");
+        REQUIRE(indexing.getNumberOfDoFs(1) == dof1Size);
+        INFO("Correct offset of second unknown");
+        REQUIRE(indexing.getDoFOffset(2) == dof1Size);
+    }
+
+    // Deassociate with an element and check
+    indexing.reinit(nullptr);
+    INFO("Without element there are no DoFs");
+    REQUIRE(indexing.getNumberOfDoFs() == 0);
+}
+
+int main(int argc, char* argv[]) {
+    // Only pass program name
+    Base::parse_options(1, argv);
+
+    int result = Catch::Session().run(argc, argv);
+
+    return result;
```

Comment:
What is the intent behind this?

Reply:
Making sure that MPI is initialized if needed. Without it there are some internals that will crash due to initialized MPI (and possibly other parts). However, I do want to pass the actual program arguments to catch.

Reply:
I've replaced to comment with a more clarifying one.

Reply:
okay nice